main/controllers/categories.py

```diff
@@ -0,0 +1,68 @@
+from typing import Annotated
+
+from fastapi import APIRouter, Depends
+
+from main.commons.exceptions import NotFound
+from main.engines.categories import (
+    add_category,
+    delete_category,
+    get_categories,
+)
+from main.schemas.base import Empty
+from main.schemas.category import CategoryCreatePayloadSchema, CategorySchema
+from main.utils.auth import require_authentication
+from main.utils.category import (
+    get_category_or_404,
+    require_category_creator,
+    validate_category_name,
+)
+
+router: APIRouter = APIRouter()
+
+
+@router.get(
+    "/categories",
+    response_model=list[CategorySchema],
+)
+async def _get_categories():
+    categories = await get_categories()
+    return categories
+
+
+@router.post(
+    "/categories",
+    response_model=CategorySchema,
+)
+async def _add_category(
+    category: CategoryCreatePayloadSchema,
+    user_id: Annotated[int, Depends(require_authentication)],
+):
+    await validate_category_name(category.name)
+    category = await add_category(name=category.name, creator_id=user_id)
+
+    if not category:
+        raise NotFound()
+
+    return category
+
+
+@router.get(
+    "/categories/{category_id}",
+    response_model=CategorySchema,
+)
+async def _get_category(category_id: int):
+    category = await get_category_or_404(category_id)
```

Comment:
Should validate the category_id before making a call to db

Reply:
defining param `category_id: int` already handle validation. 
current validation error response:
![image](https://github.com/caseykhuc/fastapi-service-learning/assets/10876529/3316ceb5-7b34-47ac-8b2c-43f85468b143)

is there another validation you're suggesting to add?

Reply:
Added positive int validation for category_id.